client/my-sites/checkout/composite-checkout/components/additional-terms-of-service-in-cart.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { TermsOfServiceRecord, useShoppingCart } from '@automattic/shopping-cart';
+import i18nCalypso, { useTranslate, TranslateResult } from 'i18n-calypso';
+import { useLocale } from '@automattic/i18n-utils';
+import { useSelector } from 'react-redux';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import Gridicon from 'calypso/components/gridicon';
+import { getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+
+const debug = debugFactory( 'calypso:composite-checkout:additional-terms-of-service' );
+
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+
+export default function AdditionalTermsOfServiceInCart(): JSX.Element | null {
+	const translate = useTranslate();
+	const locale = useLocale();
+	const { responseCart } = useShoppingCart();
+	const siteSlug = useSelector( getSelectedSiteSlug );
+
+	if ( ! responseCart.terms_of_service || responseCart.terms_of_service.length === 0 ) {
+		return null;
+	}
+
+	return (
+		<>
+			{ responseCart.terms_of_service.map( ( termsOfServiceRecord, index ) => {
+				const message = getMessageForTermsOfServiceRecord(
+					termsOfServiceRecord,
+					translate,
+					locale,
+					siteSlug
+				);
+
+				if ( ! message ) {
+					return null;
+				}
+
+				return (
+					<div className="checkout__titan-terms-of-service" key={ index }>
```

Comment:
Although in this case it's probably safe, using the array index as a React key is a risky pattern because, due to the nature of React, it's possible for the contents of an array to change, so that the indices no longer match the same data. If that happens, you can end up with unpredictable UI changes as React will lose track of which page element matches which record.

In this case it's likely that the `termsOfServiceRecord.code` is probably unique within the array, so I'd suggest that you use that as the key instead.

Reply:
I added a unique key in the backend to use it here.